lib/cocoapods-core/specification/linter.rb

```diff
@@ -387,8 +388,15 @@ def perform_github_uri_checks(git, git_uri)
       # Performs validations related to SSH sources
       #
       def check_git_ssh_source(s)
+        require 'uri'
+
         if git = s[:git]
-          if git =~ /\w+\@(\w|\.)+\:(\/\w+)*/
+          if git =~ /^#{URI.regexp}$/
```

Comment:
Can you add a test case to show what this change is for, the test you've added seems unrelated to this change since it wouldn't match this rule.